src/ui_dialogs.ts

```diff
@@ -59,6 +69,40 @@ export function makePopup(attachement: Selection<any>, title: string, body: stri
       $popup.remove();
     }
   });
+  var auto = <HTMLInputElement>$popup.select('input[autofocus]').node();
+  if (auto) {
+    auto.focus();
+  }
+  return $popup;
+
+}
+
+export function makeSortPopup(attachement: Selection<any>, title: string, body: string) {
+  var pos = offset(<Element>attachement.node());
+  var $popup = select('body').append('div')
+    .attr({
```

Comment:
the following code is highly similar to `dialogForm`. Please refactor the methods to avoid code duplication